okhttp/src/commonJvmAndroid/kotlin/okhttp3/EventListener.kt

```diff
@@ -290,6 +290,25 @@ abstract class EventListener {
   ) {
   }
 
+  /**
+   * Invoked just prior to sending bytes on the socket. Will only be invoked for upgraded connections.
+   *
+   * The connection is implicit, and will generally relate to the last [connectionAcquired] event.
```

Comment:
Can we strengthen this guarantee? Maybe we should pass the connection itself to this method?

Reply:
Similar to the checks which ensure that *End happens after *Start, leveraging this [generic check](https://github.com/square/okhttp/blob/f84fb612a455a49e3fe2c7a4516d0f2badaee708/okhttp-testing-support/src/main/kotlin/okhttp3/RecordingEventListener.kt#L148-L150)?